Canvas/SetAndResetClip.md

```diff
@@ -0,0 +1,250 @@
+# Set and Reset Clip()
+
+Authors:
+ [sushraja-msft](https://github.com/sushraja-msft),
+ [travisleithead](https://github.com/travisleithead)
+ 
+## Status of this Document
+This document is a starting point for engaging the community and standards
+bodies in developing collaborative solutions fit for standardization. As the solutions
+to problems described in this document progress along the standards-track, we will
+retain this document as an archive and use this section to keep the community up-to-date
+with the most current standards venue and content location of future work and discussions.
+* This document status: **Active**
+* Expected venue: [W3C Web Incubator Community Group](https://wicg.io/)
+* Current version: this document
+
+## Background
+
+Canvas 2d contexts support the ability to create clipping regions via the `context.clip()` API. 
+Once a clipping region is set on the canvas, subsequent drawing commands only render the 
+intersection of the drawing command's region with that of the clipping region. If an existing
+clipping region is set, calling `clip()` again will result in a clipping region that is the 
+intersection of the last clipping region with the requested region. 
+[[see spec](https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-clip)]
+
+## Clipping Concerns
+
+Four areas of concern illustrate the use cases and context for our proposal.
+
+### Animation loops
+
+When using `clip()` in an animation loop (e.g., via a `requestAnimationFrame` callback), say
+to spotlight a main character, developers will quickly realize that the "current clipping region"
+exhibits the somewhat unexpected intersection-with-previous-clip behavior. If the center of
+attention is moving, the clip region intersection behavior quickly results in an empty clipping
+region in which everything is getting clipped:
+
+```js
+let spot = { x: 50, y: 50 };
+function animationFrameCallback() {
+   context.clearRect( 0, 0, canvas.width, canvas.height );
+   let spotlight = new Path2D();
+   spotlight.arc( spot.x, spot.y, /*...*/ );
+   context.clip( spotlight ); // Create a circle clipping region centered at 'spot'
+   
+   // Draw main character here...
+   
+   // Move the spotlight to the right
+   spot.x++
+   
+   requestAnimationFrame( animationFrameCallback );
+}
+```
+
+At this point, the developer will naturally check the documentation for a way to reset the clipping
+region, but won't be able to find such an API because it doesn't exist. Eventually, they may stumble
+upon the [unintuitive workaround](https://stackoverflow.com/questions/25282793/how-to-remove-the-clip-of-a-region-in-html-5-canvas) for this limitation.
+
+Note, this use case is another argument for adding a 
```

Comment:
Can we remove this block, it dilutes our explainer and questions why not persue Clear. Clear also results in the site having to set the transform back again.

Reply:
Can do!